app/models/review.rb

```diff
@@ -0,0 +1,49 @@
+class Review < ApplicationRecord
+  belongs_to :restaurant
+  after_create :attach_review_to_restaurant
+  after_destroy :remove_review_from_restaurant
+
+  def attach_review_to_restaurant
+    restaurant = Restaurant.find(restaurant_id)
+    restaurant.rating = calc_average_add(restaurant)
```

Comment:
consider moving the `calc_average_add` to the Restaurant model, it makes more sense that the restaurant will have an `add_review` method that will also recalculate its rating and its num of reviews.
It's wired that the Review object does this for the Restaurant.

Reply:
I agree with Dotan, and I want to suggest another option.
Since often the models get really big really fast, it might be useful to extract certain things. Since there's a lot of rating calculation logic here you could create a rating calculator to handle all of this.

Reply:
Agreed, I thought about it too but from some reason didn't do it.
Done